contracts/rust/deployer/src/lib.rs

```diff
@@ -1763,4 +1879,187 @@ mod tests {
+        let mut multisig_admin = Address::random();
+        let mut mnemonic =
+            "test test test test test test test test test test test junk".to_string();
+        let mut account_index = 0;
+        let anvil = Anvil::default().spawn();
+        let dry_run = matches!(options.run_mode, RunMode::DryRun);
+
+        if !dry_run {
+            dotenvy::from_filename_override(".env.deployer.rs.test").ok();
+
+            for item in dotenvy::from_filename_iter(".env.deployer.rs.test")
+                .expect("Failed to read .env.deployer.rs.test")
+            {
+                let (key, val) = item?;
+                if key == "ESPRESSO_SEQUENCER_L1_PROVIDER" {
+                    sepolia_rpc_url = val.to_string();
+                } else if key == "ESPRESSO_SEQUENCER_ETH_MULTISIG_ADDRESS" {
+                    multisig_admin = val.parse::<Address>()?;
+                } else if key == "ESPRESSO_SEQUENCER_ETH_MNEMONIC" {
+                    mnemonic = val.to_string();
+                } else if key == "ESPRESSO_DEPLOYER_ACCOUNT_INDEX" {
+                    account_index = val.parse::<u32>()?;
+                }
+            }
+
+            if sepolia_rpc_url.is_empty() || multisig_admin.is_zero() {
+                panic!("ESPRESSO_SEQUENCER_L1_PROVIDER and ESPRESSO_SEQUENCER_ETH_MULTISIG_ADDRESS must be set in .env.deployer.rs.test");
+            }
+        }
+
+        let mut contracts = Contracts::new();
+        let admin_signer = MnemonicBuilder::<English>::default()
+            .phrase(mnemonic)
+            .index(account_index)
+            .expect("wrong mnemonic or index")
+            .build()?;
+        let admin = admin_signer.address();
+        let provider = if !dry_run {
+            ProviderBuilder::new()
+                .wallet(admin_signer)
+                .connect(&sepolia_rpc_url)
+                .await?
+        } else {
+            ProviderBuilder::new()
+                .wallet(admin_signer)
+                .connect(&anvil.endpoint())
+                .await?
+        };
+        let init_recipient = provider.get_accounts().await?[0];
+        let initial_supply = U256::from(3590000000u64);
+        let token_name = "Espresso";
+        let token_symbol = "ESP";
+
+        // deploy proxy and V1
+        let esp_token_proxy_addr = deploy_token_proxy(
+            &provider,
+            &mut contracts,
+            admin,
+            init_recipient,
+            initial_supply,
+            token_name,
+            token_symbol,
+        )
+        .await?;
+        if matches!(options.upgrade_count, UpgradeCount::Twice) {
+            // upgrade to v2
+            upgrade_esp_token_v2(&provider, &mut contracts).await?;
+        }
+
+        // transfer ownership to multisig
+        let _receipt = transfer_ownership(
+            &provider,
+            Contract::EspTokenProxy,
+            esp_token_proxy_addr,
+            multisig_admin,
+        )
+        .await?;
```

Comment:
Our transfer ownership function should probably check that the receipt status is 1. Currently I think we don't do that. Don't need to fix it in this PR though.